### PR TITLE
feat: add AAD support in aes gcm

### PR DESCRIPTION
**Pull request by wzxjohn**
#98

### CODE REVIEW

**Comment by wzxjohn** [comment]
@zhuizhuhaomeng Sorry for accidentally merge the master into patch. I close the old PR and start this new one.

**Comment by wzxjohn** [comment]
@zhuizhuhaomeng test case added

File: t/aes.t [hunk 1]
```diff
@@ -561,3 +561,27 @@ AES-256 CBC (custom keygen, without user padding, enable padding) HEX: 794617717
 true
 --- no_error_log
 [error]
+
```
Comment (zhuizhuhaomeng) [inline t/aes.t:564]: request three blank lines here.

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

File: lib/resty/aes.lua [hunk 2]
```diff
@@ -241,6 +241,14 @@ function _M.encrypt(self, s)
         return nil, "EVP_EncryptInit_ex failed"
     end
 
+    if self._cipher == "gcm" then
+        if aad ~= nil then
+            if C.EVP_EncryptUpdate(ctx, nil, tmp_len, aad, #aad) == 0 then
+                return nil, "C.EVP_EncryptUpdate failed"
+            end
+        end
+    end
```
Comment (zhuizhuhaomeng) [inline lib/resty/aes.lua:250]:
```suggestion
    if self._cipher == "gcm" and aad ~= nil then
        if C.EVP_EncryptUpdate(ctx, nil, tmp_len, aad, #aad) == 0 then
            return nil, "C.EVP_EncryptUpdate failed"
        end
    end
```

File: lib/resty/aes.lua [hunk 3]
```diff
@@ -284,6 +292,14 @@ function _M.decrypt(self, s, tag)
       return nil, "EVP_DecryptInit_ex failed"
     end
 
+    if self._cipher == "gcm" then
+        if aad ~= nil then
+            if C.EVP_DecryptUpdate(ctx, nil, tmp_len, aad, #aad) == 0 then
+                return nil, "C.EVP_DecryptUpdate failed"
+            end
+        end
+    end
```
Comment (zhuizhuhaomeng) [inline lib/resty/aes.lua:301]:
```suggestion
    if self._cipher == "gcm" and aad ~= nil then
            if C.EVP_DecryptUpdate(ctx, nil, tmp_len, aad, #aad) == 0 then
                return nil, "C.EVP_DecryptUpdate failed"
            end
    end
```

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

**Comment by zhuizhuhaomeng** [comment]
merge with small modification.

**Pull request closed without merging**